centreon/src/Centreon/Infrastructure/DatabaseConnection.php

```diff
@@ -21,96 +22,862 @@
-            }
+            parent::__construct(
+                $this->connectionConfig->getMysqlDsn(),
+                $this->connectionConfig->getUser(),
+                $this->connectionConfig->getPassword(),
+                [
+                    \PDO::MYSQL_ATTR_INIT_COMMAND => "SET NAMES {$this->connectionConfig->getCharset()}",
+                    \PDO::ATTR_ERRMODE => \PDO::ERRMODE_EXCEPTION,
+                    \PDO::ATTR_DEFAULT_FETCH_MODE => \PDO::FETCH_ASSOC,
+                ]
+            );
+        } catch (\PDOException $exception) {
+            $this->writeDbLog(
+                message: "Unable to connect to database : {$exception->getMessage()}",
+                customContext: ['dsn_mysql' => $this->connectionConfig->getMysqlDsn()],
+                previous: $exception,
+            );
+
+            throw ConnectionException::connectionFailed($exception);
         }
     }
 
     /**
+     * Factory
+     *
+     * @param ConnectionConfig $connectionConfig
+     *
+     * @throws ConnectionException
+     * @return DatabaseConnection
+     */
+    public static function createFromConfig(ConnectionConfig $connectionConfig): self
+    {
+        return new self($connectionConfig);
+    }
+
+    /**
+     * switch connection to another database
+     *
+     * @param string $dbName
+     *
+     * @throws ConnectionException
+     */
+    public function switchToDb(string $dbName): void
+    {
+        $this->executeStatement('use ' . $dbName);
+    }
+
+    /**
+     * To get the used native connection by DBAL (PDO, mysqli, ...).
+     *
+     * @return \PDO
+     */
+    public function getNativeConnection(): \PDO
+    {
+        return $this;
+    }
+
+    /***
+     * Returns the ID of the last inserted row.
+     * If the underlying driver does not support identity columns, an exception is thrown.
+     *
+     * @throws ConnectionException
      * @return string
      */
-    public function getCentreonDbName()
+    public function getLastInsertId(): string
     {
-        return $this->centreonDbName;
+        try {
+            return (string) $this->lastInsertId();
+        } catch (\Throwable $exception) {
+            $this->writeDbLog(
+                message: 'Unable to get last insert id',
+                previous: $exception,
+            );
+
+            throw ConnectionException::getLastInsertFailed($exception);
+        }
     }
 
     /**
-     * @param string $centreonDbName
+     * Check if a connection with the database exist.
+     *
+     * @return bool
      */
-    public function setCentreonDbName(string $centreonDbName): void
+    public function isConnected(): bool
     {
-        $this->centreonDbName = $centreonDbName;
+        try {
+            $this->executeSelectQuery('SELECT 1');
+
+            return true;
+        } catch (ConnectionException $exception) {
+            $this->writeDbLog(
+                message: 'Unable to check if the connection is established',
```

Comment:
We were not unable to check: we were able to check and the result is that we are not connected

```suggestion
                message: 'Unable to establish the connection.',
```